packages/conf/src/types.ts

```diff
@@ -62,12 +62,12 @@ export type CatalogFormatter = {
   templateExtension?: string
   parse(
     content: string,
-    ctx: { locale: string | null; sourceLocale: string; filename: string }
+    ctx: { locale: string; sourceLocale: string; filename: string }
```

Comment:
from the call site, it looks like the `locale` is always present, while `existing` may not.

Reply:
locale may not be present in case if you generating a template.

Reply:
please return `| null` back